### PR TITLE
Extended the message type so can make an example application on ci40 with the python API

### PR DESCRIPTION
**Pull request by yi-liang**
In order to make a python version of https://docs.creatordev.io/ci40/iotkit/workshop-1-switchcounter/
(1) Added Python3 support as running python3 on ci40 image;
(2) Corrected some debug info
(3) Added define message so it can create an object from the application
Please find the example application on https://github.com/yi-liang/creator_py_examples/blob/master/workshop1_awaApi.py

### CODE REVIEW

**Comment by yi-liang** [comment]
Hi @GonZo, would the CI run a daily build? there was one test failed but doesn't look like caused by python,not sure if all tests will pass at the moment? I tried to look for the last pull request/test results, seems there were 67 tests (all passed), but look like this time it was trying to run 1965 tests(1 failed) ?

> TestWriteValueArray1/TestWriteValueArray.TestWriteValueArray/29, where GetParam() = Item: expectedAddResult 7, expectedProcessResult 9, UseSetOperation 1, PopulateArrayWithInitialValues 0, objectID 65534, objectInstanceID 0, resourceID 4, value 0x10ed750, valueCount 3, type12

File: api/python/ipc_core.py [hunk 1]
```diff
@@ -13,18 +13,27 @@
...
 IPC Core Functionality
 """
 
-from lxml import etree
+# py-lxml is not yet available in openwrt
+#from lxml import etree
+import xml.etree.ElementTree as etree
+
+# Check python version
+from sys import version_info
```
Comment (mtusnio) [inline api/python/ipc_core.py]: According to this - http://python-future.org/compatible_idioms.html - the way to fix the iteritems/items discrepancy is by doing an import:
from future.utils import viewitems

File: api/python/ipc_core.py [hunk 2]
```diff
@@ -306,8 +315,12 @@ def __str__(self):
             s += "value %s, " % (self._value,)
         if self._children:
             s += "children "
-            for key, value in self._children.iteritems():
-                s += str(key) + ":%x" % (id(value),) + " "
+            if PY_VERSION == 3:
+                for key, value in self._children.items():
```
Comment (mtusnio) [inline api/python/ipc_core.py]: viewitems(self._children) if the above import is done

**Comment by boyvinall** [comment]
Closing now, since #324 was merged.

**Pull request closed without merging**